06_puppy/srini/Types.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+//PuppyRecord is internal type with embedded puppy
+type PuppyRecord struct {
+	id    int
+	puppy Puppy
+}
+
+//Puppy is struct for Puppy bread
+type Puppy struct {
+	breed  string
+	colour string
+	value  float64
+}
+
+//Storer provides CRUD methods
+type Storer interface {
+	CreatePuppy(*Puppy) int
```

Comment:
I'd still keep this example pointer free. Puppy is very small and working without pointers simplify some things imo.